actions.md

```diff
@@ -0,0 +1,547 @@
+| 2019-03-16 20:02:57 | 22da217c8e3a4b799a217c8e3a0b79c4 | sequence | warm | 3.46s | success | guest/tripleAndIncrement:0.0.1
+</pre>
+
+Here's the meaning of each column in the list:
+
+| Column | Description |
+| :--- | :--- |
+| `Datetime` | The date and time when the invocation occurred. |
+| `Activation ID` | An activation ID that can be used to retrive the result using the `nim activation get`, `nim activation result` and `nim activation logs` commands. |
+| `Kind` | The runtime or action type |
+| `Start` | An indication of the latency, i.e. if the runtime container was cold or warm started. |
+| `Duration` | Time taken to execute the invocation. |
+| `Status` | The outcome of the invocation. For an explanation of the various statuses, see the description of the `statusCode` below. |
+| `Entity` | The fully qualified name of entity that was invoked. |
+
+## Further considerations for creating actions
+
+- An action executes in a sandboxed environment, namely a container. At any given time, a single activation will execute inside the container. Subsequent invocations of the same action may reuse a previous container, and there may exist more than one container at any given time, each having its own state.
+- Invocations of an action are not ordered. If the user invokes an action twice from the command line or the REST API, the second invocation might run before the first. If the actions have side effects, they might be observed in any order.
+- There is no guarantee that actions will execute atomically. Two actions can run concurrently and their side effects can be interleaved. `nim` does not ensure any particular concurrent consistency model for side effects. Any concurrency side effects will be implementation-dependent.
+- Actions have two phases: an initialization phase, and a run phase. During initialization, the function is loaded and prepared for execution. The run phase receives the action parameters provided at invocation time. Initialization is skipped if an action is dispatched to a previously initialized container. This is referred to as a _warm start_. You can tell if an [invocation was a warm activation or a cold one requiring initialization](annotations.md#annotations-specific-to-activations) by inspecting the activation record.
+- An action runs for a bounded amount of time. This limit can be configured per action, and applies to both the initialization and the execution separately. If the action time limit is exceeded during the initialization or run phase, the activation's response status is _action developer error_.
+
+## Other action commands
+
+### Get metadata for actions
+
+Metadata that describes existing actions can be retrieved via the `nim action get` command.
+
+**[[NH: check output, looks like the original had no package so my command is wrong?]]**
+```
+nim action get greeting
+ok: got action greeting
```

Comment:
`nim` does not put a tag line like this; currently, the output starts with the JSON